data.py

```diff
@@ -0,0 +1,19 @@
+from csv import reader as csv_reader
+
+from questions import Questions
+
+
+class Source:
+
+    @staticmethod
+    def from_csv(path: str) -> [Questions]:
+        questions = []
+        with open(path, 'r', encoding='utf-8') as f:
+            csv = csv_reader(f)
```

Comment:
csv.reader is only for the file? Try to make this function is working with [byte], and create another function called from_csv_file.

So you don't have to hardcode your testing file in the argument.

Reply:
like this?
```python
@staticmethod
    def from_csv(path: str):
        content = ''
        with open(path, 'rt', encoding='utf-8') as f:
            content = f.read()
        return csv.reader(io.StringIO(content))

```